aws_secretsmanager error calling aws_secretsmanager: Error getting secret: MissingRegion

HI, i have that error: 

```
2020-05-15T01:49:05.9443153Z ##[section]Starting: Packer build
2020-05-15T01:49:05.9455589Z ==============================================================================
2020-05-15T01:49:05.9456168Z Task         : Packer
2020-05-15T01:49:05.9456652Z Description  : Build Automated Machine Images
2020-05-15T01:49:05.9457125Z Version      : 1.185.1444
2020-05-15T01:49:05.9457565Z Author       : Manuel Riezebosch
2020-05-15T01:49:05.9458127Z Help         : [More information](https://packer.io)
2020-05-15T01:49:05.9458783Z ==============================================================================
2020-05-15T01:49:06.1006801Z [command]/vsts_agent/agent/_work/_tool/packer/1.5.6/x64/packer build -var OWNER_SOURCE=763669947983 -var AMI_OWNER=813024095984 -var passwortest=*** -var region="us-east-1" -var 'region=us-east-1' -color=false /vsts_agent/agent/_work/r4/a/_aw0499001_bb7_aws_packer/pipeline/linux_base_was_release-testsc.json
2020-05-15T01:49:06.1625263Z Error initializing core: template: root:1:3: executing "root" at <aws_secretsmanager `bb4certificados`>: error calling aws_secretsmanager: Error getting secret: MissingRegion: could not find region configuration
2020-05-15T01:49:06.1626590Z 
2020-05-15T01:49:06.1630922Z ==> Builds finished but no artifacts were created.
2020-05-15T01:49:06.2854915Z 
2020-05-15T01:49:06.2989345Z ##[error]Error: The process '/vsts_agent/agent/_work/_tool/packer/1.5.6/x64/packer' failed with exit code 1
2020-05-15T01:49:06.3005843Z ##[section]Finishing: Packer build
```

i anex the json template

```
{
  "variables": {
    "access_key": "",
    "security_token": "",
    "aws_secret_key": "{{env `AWS_SECRET_ACCESS_KEY`}}",
    "aws_iam_profile_name": "PackerProfile",
    "aws_region": "{{env `AWS_DEFAULT_REGION`}}",
    "region":"{{env `AWS_DEFAULT_REGION`}}",
    "vpc_id": "vpc-079d901fef225a775",
    "subnet_id": "subnet-0b03aefb2c7d4046b",
    "password": "{{ aws_secretsmanager `bb4certificados` }}"
  },
  "sensitive-variables": [
    "aws_access_key",
    "aws_secret_key",
    "passwortest",
    "password"
  ],
  "builders": [
    {
      "type": "amazon-ebs",
      "access_key": "{{user `access_key`}}",
      "secret_key": "{{user `secret_key`}}",
      "iam_instance_profile": "{{user `aws_iam_profile_name`}}",
      "region": "{{user `aws_region`}}",
      "vpc_id": "{{user `vpc_id`}}",
      "subnet_id": "{{user `subnet_id`}}",
      "associate_public_ip_address": true,
      "source_ami_filter": {
        "filters": {
          "name": "IBM-WAS-FULL-P-Rhel7.7-*"
        },
        "owners": [
          "{{user `OWNER_SOURCE`}}"
        ],
        "most_recent": true
      },
      "instance_type": "t2.2xlarge",
      "ssh_username": "ec2-user",
      "ami_name": "BB7-APP-WAS-RELEASE-test-{{user `AMBIENTE`}}-{{timestamp}}",
      "ami_users": [       
        "{{user `AMI_OWNER`}}"       
      ],
      
    
  ],
  "provisioners": [
    {
      "type": "file",
      "source": "{{ template_dir }}/assets/was/packer_deploy1.sh",
      "destination": "/tmp/assets/"
    },
    {
      "type": "shell",
      "pause_before": "5s",
      "environment_vars": [
        "AWS_ACCESS_KEY_ID={{user `access_key`}}",
        "AWS_SECRET_ACCESS_KEY={{user `secret_key`}}",
        "AWS_DEFAULT_REGION={{user `aws_region`}}",
        "AWS_DEFAULT_REGION={{user `AWS_DEFAULT_REGION`}}",
        "AWS_IAM_PROFILE_NAME={{user `AWS_IAM_PROFILE_NAME`}}",
        "AWS_VPC_ID={{user `AWS_VPC_ID`}}",
        "AWS_SUBNET_ID={{user `AWS_SUBNET_ID`}}",
        "testpwd={{user `passwortest`}}",
        "testpwdSM={{user password``}}"
      ],
      
      "inline": [     
        "echo $testpwd > /var/tmp/passAWS",
        "echo $testpwdSM > /var/tmp/passAWSSM",
        "sudo bash -x /tmp/assets/packer_deploy1.sh $testpwd $testpwdSM"
      ]
    } 
  ],
  "post-processors": [
    [
      {
        "output": "manifest.json",
        "strip_path": true,
        "type": "manifest"
      }
    ]
  ]
}
```

--- a/template/interpolate/aws/secretsmanager/secretsmanager.go
+++ b/template/interpolate/aws/secretsmanager/secretsmanager.go
@@ -5,6 +5,7 @@ package secretsmanager
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
@@ -31,14 +32,15 @@ func New(config *AWSConfig) *Client {
 
 func (c *Client) newSession(config *AWSConfig) *session.Session {
 	// Initialize config with error verbosity
-	sess := aws.NewConfig().WithCredentialsChainVerboseErrors(true)
+	sessConfig := aws.NewConfig().WithCredentialsChainVerboseErrors(true)
 
 	if config.Region != "" {
-		sess = sess.WithRegion(config.Region)
+		sessConfig = sessConfig.WithRegion(config.Region)
 	}
 
 	opts := session.Options{
-		Config: *sess,
+		SharedConfigState: session.SharedConfigEnable,
+		Config:            *sessConfig,
 	}
 
 	return session.Must(session.NewSessionWithOptions(opts))
@@ -102,5 +104,5 @@ func getSecretValue(s *SecretString, spec *SecretSpec) (string, error) {
 		return v, nil
 	}
 
-	return "", errors.New("No secret found")
+	return "", fmt.Errorf("No secret found for key %q", spec.Key)
 }

--- a/template/interpolate/funcs.go
+++ b/template/interpolate/funcs.go
@@ -299,12 +299,12 @@ func funcGenAwsSecrets(ctx *Context) interface{} {
 		if !ctx.EnableEnv {
 			// The error message doesn't have to be that detailed since
 			// semantic checks should catch this.
-			return "", errors.New("AWS Secrets Manager vars are only allowed in the variables section")
+			return "", errors.New("AWS Secrets Manager is only allowed in the variables section")
 		}
 
 		// Check if at least 1 parameter has been used
 		if len(secret) == 0 {
-			return "", errors.New("At least one parameter must be used")
+			return "", errors.New("At least one secret name must be provided")
 		}
 		// client uses AWS SDK CredentialChain method. So,credentials can
 		// be loaded from credential file, environment variables, or IAM
@@ -329,7 +329,7 @@ func funcGenAwsSecrets(ctx *Context) interface{} {
 
 		s, err := client.GetSecret(spec)
 		if err != nil {
-			return "", fmt.Errorf("Error getting secret: %s", err)
+			return "", err
 		}
 		return s, nil
 	}

